src/nano-network-api.js

```diff
@@ -207,45 +233,67 @@ export class NanoNetworkApi {
-                    const agent = new Nimiq.NetworkAgent(consensus.blockchain, consensus.network.addresses, networkConfig, channel);
-                    let header = null;
-                    channel.on('head', (msg) => {
-                        header = msg.header;
-                        console.debug(`[Pico] Current height is ${header.height}`);
-                        onChannelHead(channel, header);
-                    });
-                    channel.on('accounts-proof', (msg) => {
-                        onBalancesMsg(msg);
-                    })
-                    agent.on('handshake', () => {
-                        channel.getHead();
-                    });
+            function startPicoOnChannel(channel) {
+                // Note that we stop the pico consensus checks once a consensus was reached (either by pico or nano
+                // fallback). However, during nano fallback, we keep them alive as we might still get to a pico
+                // consensus before the nano consensus.
+                channel.on('head', (msg) => {
+                    if (resolved) return;
+                    const header = msg.header;
+                    console.debug(`[Pico] Current height is ${header.height}`);
+                    onChannelHead(channel, header);
+                });
+                channel.on('accounts-proof', (msg) => {
+                    if (resolved) return;
+                    onBalancesMsg(msg);
                 });
+                channel.getHead();
+            }
 
-                // Testnet only has 4 seeds
-                connector.connect(Nimiq.GenesisConfig.SEED_PEERS[i]);
+            for (const channel of establishedChannels) {
+                startPicoOnChannel(channel);
+            }
+
+            if (establishedChannels.length < 4) {
+                network.on('peer-joined', (peer) => {
+                    if (resolved || Nimiq.Services.isNanoNode(peer.peerAddress.services)) return;
+                    startPicoOnChannel(peer.channel);
+                });
+
+                let additionalChannelsToConnectTo = 4 - establishedChannels.length;
+                for (const peerAddress of Nimiq.GenesisConfig.SEED_PEERS) {
+                    if (additionalChannelsToConnectTo <= 0) break;
+                    if (!network.connections.connectOutbound(peerAddress)) continue; // continue on duplicate connection
+                    --additionalChannelsToConnectTo;
+                }
             }
 
             setTimeout(() => {
-                if (resolved) return;
+                if (resolved || usingFallback) return;
                 fallbackToNanoConsensus();
             }, 5000);
         });
+
+        establishedChannels.splice(0, establishedChannels.length);
+        picoHeads.splice(0, picoHeads.length);
+        currentHead = null;
+        picoChannels.splice(0, picoChannels.length);
+        picoBalances.clear();
```

Comment:
Why do these variables need to be reset (and why with `splice()`)? Can you add a comment above the reset block?

Reply:
I think they are still referenced from the scopes of the event handlers, thus would result in a small memory leak. The proper thing to do would be to unbind the event handlers, I was just too lazy to do that :smile: 
But yeah, should actually do it.
I used splice because I didn't want to change the variables from being const.